### ISSUE TITLE
Amp form always show submit-success in Gmail

### ISSUE THREAD

**Issue by 06romix** [ampproject/amphtml#24774]
## amp-form [submit-error] doesn't work correctly in Gmail

Form in witch `<div submit-success>` doesn't include tag `<template type="amp-mustache"></template>` always show success message whether submit-error include `<template type="amp-mustache">` or not.

```html
<!doctype html>
<html ⚡4email>
<head>
  <meta charset="utf-8">
  <script async src="https://cdn.ampproject.org/v0.js"></script>
  <script async custom-element="amp-form" src="https://cdn.ampproject.org/v0/amp-form-0.1.js"></script>
  <script async custom-template="amp-mustache" src="https://cdn.ampproject.org/v0/amp-mustache-0.2.js"></script>
  <style amp4email-boilerplate>body{visibility:hidden}</style>
  <style amp-custom>
    body {
      padding-bottom: 40px;
    }
    h1 {
      margin: 1rem;
    }
  </style>
</head>
<body>
<form method="post"
  action-xhr="https://ampbyexample.com/components/amp-form/submit-form-input-text-xhr">
  <p>Form Submission with Page Update</p>
  <div>
    <input type="text"
      name="name"
      placeholder="Name..."
      required>
  </div>
  <input type="submit"
    value="Subscribe">
  <div submit-success>
      Success! Thank you for trying the
      <code>amp-form</code> demo! Try to insert the word "error" as a name input in the form to see how
      <code>amp-form</code> handles errors.
  </div>
  <div submit-error>
    <template type="amp-mustache">
      Error! Thanks {{name}} for trying the
      <code>amp-form</code> demo with an error response.
    </template>
  </div>
</form>
</body>
</html>
```

## How do we reproduce the issue?

1. Send code above from https://amp.gmail.dev/playground/ 
2. Go to gmail
3. Fill form with "error" string
4. Send form
5. See "Success! Thank you for trying..." instead of "Error! Thanks error for trying the ..."

## What browsers are affected?
On Ubuntu

Google Chrome Version 77.0.3865.90 (Official Build) (64-bit)
Firefox 69.0.1

## Which AMP version is affected?

I think it was always broken
Powered by AMP ⚡ HTML – Version 1909141409590
